spec.bs

```diff
@@ -468,7 +479,7 @@ A <dfn>component</dfn> is a [=struct=] with the following [=struct/items=]:
 </div>
 
 <div algorithm>
-  To perform a <dfn export for="URL pattern">match</dfn> given a [=URL pattern=] |urlPattern|, a {{URLPatternInput}} |input|, and an optional string |baseURLString|:
+  To perform a <dfn export for="URL pattern">match</dfn> given a [=URL pattern=] |urlPattern|, a {{URLPatternMatchInput}} |input|, and an optional string |baseURLString|:
```

Comment:
This modification is not correct, because it accepts a `URL` JavaScript object. You want to accept a [URL](https://url.spec.whatwg.org/#concept-url) struct.

Reply:
I assume we could change `a {{URLPatternInput}} |input|` to `a {{URLPatternInput}} or [=/url=] |input|`. Is that a right approach?

Reply:
Yeah, exactly

Reply:
Let me adjust the title and description for the direction.
I intend to make the JavaScript APIs accept a URL object, and this PR is for that.  However, I actually do not need that for the SW static routing API.

Reply:
Adjusted the title and the description.  I will change the spec aligned with that.

Reply:
done.
@jeremyroman In https://github.com/whatwg/urlpattern/issues/218#issuecomment-1946247413, you mentioned "a spec helper which accepts URL records".  If you prefer the separated algorithm to accept [=/URL=], please feel free to say so.  I am happy to change.